model/user/user.go

```diff
@@ -88,6 +88,32 @@ func (u *DBUser) AddPublicKey(keyName, keyValue string) error {
 	return nil
 }
 
+func (u *DBUser) DeletePublicKey(keyName string) error {
+	selector := bson.M{
+		IdKey: u.Id,
+		bsonutil.GetDottedKeyName(PubKeysKey, PubKeyNameKey): bson.M{"$eq": keyName},
+	}
+	update := bson.M{
+		"$pull": bson.M{
+			PubKeysKey: bson.M{
+				PubKeyNameKey: keyName,
+			},
+		},
+	}
+	if err := UpdateOne(selector, update); err != nil {
+		return err
+	}
+
+	newKeys := []PubKey{}
```

Comment:
I might be wrong about this but it seems like you can do this without declaring a separate variable. 

Maybe it makes more sense to use findAndModify for this?